cmd/cloud-controller-manager/nodeipamcontroller.go

```diff
@@ -55,7 +55,7 @@ type nodeIPAMController struct {
 func (nodeIpamController *nodeIPAMController) startNodeIpamControllerWrapper(completedConfig *cloudcontrollerconfig.CompletedConfig, cloud cloudprovider.Interface) app.InitFunc {
 	errors := nodeIpamController.nodeIPAMControllerOptions.Validate()
 	if len(errors) > 0 {
-		klog.Fatal("NodeIPAM controller values are not properly set.")
+		klog.ErrorS("NodeIPAM controller values are not properly set.")
```

Comment:
Suggestion: 
klog.ErrorS(utilerrors.NewAggregate(errors), "NodeIPAM controller values are not properly set.")
OR
klog.ErrorS(nil, "NodeIPAM controller values are not properly set.")

Prefer the first one to keep the error details.

Reply:
utilerrors import fyi:
utilerrors "k8s.io/apimachinery/pkg/util/errors"

Reply:
Not convinced that moving away from fatal is a good idea. I realize this is "sample" code but if the real version of this code has an error you have a non-functional cluster and your making it more difficult to debug.